pkg/util/ecs/common.go

```diff
@@ -0,0 +1,171 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build docker
+
+package ecs
+
+import (
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/docker"
+	log "github.com/cihub/seelog"
+)
+
+const (
+	metadataURL string = "http://169.254.170.2/v2/metadata"
+	statsURL    string = "http://169.254.170.2/v2/stats"
+	timeout            = 500 * time.Millisecond
+)
+
+// GetTaskMetadata extracts the metadata payload for the task the agent is in.
+func GetTaskMetadata() (TaskMetadata, error) {
+	var meta TaskMetadata
+	client := http.Client{
+		Timeout: timeout,
+	}
+	resp, err := client.Get(metadataURL)
+	if err != nil {
+		return meta, err
+	}
+	defer resp.Body.Close()
+
+	decoder := json.NewDecoder(resp.Body)
+	err = decoder.Decode(&meta)
+	if err != nil {
+		log.Errorf("decoding task metadata failed - %s", err)
+	}
+	return meta, err
+}
+
+// GetECSContainers returns all containers exposed by the ECS API as plain ECSContainers
+func GetECSContainers() ([]Container, error) {
+	meta, err := GetTaskMetadata()
+	if err != nil || len(meta.Containers) == 0 {
+		log.Errorf("unable to retrieve task metadata")
```

Comment:
Trying to clean up some noisy logging in the process agent at the moment. Can we remove this log, and just return the `err`? This gets logged by its upstream caller in (`GetContainers()` in providers.go)

Reply:
For context, if the user does not have any listeners configured, we're setting both `ecs` and `docker` listeners as a default in the process-agent.

It's likely that ECS will not be enabled on most setups, though that means it'll log quite a few errors at the moment. I've copied & modified `providers.go` on our clean up the logging a bit for now but this remains: https://github.com/DataDog/datadog-process-agent/pull/93/files#diff-26a7c4d5ec4700a56067f3a088d2de55R32